src/AerospikeCache.php

```diff
@@ -0,0 +1,133 @@
+        parent::__construct($cacheNamespace, $defaultLifetime);
+    }
+
+    protected function doFetch(array $ids): array
+    {
+        $result = [];
+
+        $keys = $this->initializeKeysForAerospike($ids);
+
+        $this->aerospike->getMany($keys, $records);
+
+        foreach ($records as $record) {
+            if ($record['metadata'] !== null) {
+                $result[$record['key']['key']] = isset($record['bins'][self::WRAPPER_NAME]) ?
+                    $record['bins'][self::WRAPPER_NAME] : null;
+            }
+        }
+
+        return $result;
+    }
+
+    private function initializeKeysForAerospike(array $ids): array
+    {
+        return array_reduce(
+            $ids,
+            function (array $values, $id) {
+                $values[] = $this->createKey($id);
+
+                return $values;
+            },
+            []
+        );
+    }
+
+    protected function doHave($id)
+    {
+        $statusCode = $this->aerospike->get($this->createKey($id), $read);
+
+        return $statusCode === \Aerospike::OK;
+    }
+
+    protected function doClear($namespace = ''): bool
+    {
+        if ($namespace === '') {
+            $statusCode = $this->aerospike->truncate($this->namespace, $this->set, 0);
+        } else {
+            $clearNamespace = function ($record) use ($namespace): void {
+                if ($namespace === mb_substr($record['key']['key'], 0, mb_strlen($namespace))) {
+                    $status = $this->aerospike->remove($record['key']);
+                    if (!$this->isStatusOkOrNotFound($status)) {
+                        throw new CacheException($this->aerospike->error());
+                    }
+                }
+            };
+
+            $this->aerospike->scan($this->namespace, $this->set, $clearNamespace);
+
+            $statusCode = 0;
+        }
+
+        return $this->isStatusOkOrNotFound($statusCode);
+    }
+
+    protected function doDelete(array $ids): bool
+    {
+        $removedAllItems = true;
+
+        foreach ($ids as $id) {
+            $statusCode = $this->aerospike->remove($this->createKey($id));
+            if (!$this->isStatusOkOrNotFound($statusCode)) {
+                $removedAllItems = false;
+            }
+        }
+
+        return $removedAllItems;
+    }
+
+    protected function doSave(array $values, $lifetime): bool
+    {
+        foreach ($values as $key => $value) {
+            $data = [self::WRAPPER_NAME => $value];
+            $statusCode = $this->aerospike->put($this->createKey($key), $data, $lifetime, [\Aerospike::OPT_POLICY_KEY => \Aerospike::POLICY_KEY_SEND]);
+        }
+
+        return $statusCode === \Aerospike::OK;
+    }
+
+    private function createKey($key)
+    {
+        return $this->aerospike->initKey($this->namespace, $this->set, $key);
+    }
+
+    private function isStatusOkOrNotFound(int $statusCode): bool
+    {
+        if ($statusCode === \Aerospike::ERR_RECORD_NOT_FOUND || $statusCode === \Aerospike::OK) {
```

Comment:
You can simply return the result (see https://www.tomasvotruba.cz/blog/2018/05/21/is-your-code-readable-by-humans-cognitive-complexity-tells-you/#1-refactoring-to-shorter-condition)
```suggestion
        return $statusCode === \Aerospike::ERR_RECORD_NOT_FOUND || $statusCode === \Aerospike::OK;
```

Reply:
That is right thanks!